internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2531,6 +2568,19 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 	return resourceKubernetesClusterRead(d, meta)
 }
 
+func validateKubernetesClusterUpgradeOverrideSetting(_, new interface{}) error {
+	if new == nil {
+		return fmt.Errorf("upgrade_override cannot be unset")
```

Comment:
Can we wrap the field in an back ticks here?

```suggestion
		return fmt.Errorf("`upgrade_override` cannot be unset")
```